app/javascript/packs/admin_shifts.js

```diff
@@ -267,72 +294,76 @@ const hideShowEvents = (eventName, toggleId, text) => {
   } else {
     sourceShow[eventName] = "none";
   }
-  [...document.getElementsByClassName("shift-hide-button")].forEach((el) => {
-    el.innerText = `${sourceShow[eventName] === "block" ? "Show" : "Hide"}`;
-  });
+  if (eventName === "transparent") {
+    [...document.getElementsByClassName("shift-hide-button")].forEach((el) => {
+      el.checked = sourceShow[eventName] === "none";
+    });
+  }
 };
 
 document
   .getElementById("hide-show-unavailabilities")
   .addEventListener("click", () => {
-    hideShowEvents(
-      "transparent",
-      "hide-show-unavailabilities",
-      "Unavailabilities"
-    );
+    hideShowEvents("transparent");
   });
 
 document
   .getElementById("hide-show-google-events")
   .addEventListener("click", () => {
-    hideShowEvents("google", "hide-show-google-events", "Event");
+    hideShowEvents("google");
   });
 
 document
   .getElementById("hide-show-staff-needed")
   .addEventListener("click", () => {
-    hideShowEvents("staffNeeded", "hide-show-staff-needed", "Staff Needed");
+    hideShowEvents("staffNeeded");
   });
 
-const start_picker = start_datetime.flatpickr({
-  enableTime: true,
-  time_24hr: true,
-  altInput: true,
-  altFormat: "F j, Y at H:i",
-});
-
-const end_picker = end_datetime.flatpickr({
-  enableTime: true,
-  time_24hr: true,
-  altInput: true,
-  altFormat: "F j, Y at H:i",
-});
-
-modalClose.addEventListener("click", closeModal);
-
-modalSave.addEventListener("click", () => {
-  createCalendarEvent();
-});
-
-window.toggleVisibility = (name) => {
-  document.getElementById(name).innerText = `${
-    document.getElementById(name).innerText == "Hide" ? "Show" : "Hide"
-  }`;
-  let staff_name = document.getElementById(name).dataset.staffname;
+// Toggle Staff Visibility
+window.toggleVisibility = (id) => {
   let allEvents = calendar.getEvents();
   for (let ev of allEvents) {
-    if (ev.title.startsWith(staff_name)) {
+    if (ev.extendedProps.userId === id) {
       ev.setProp(
         "display",
-        document.getElementById(name).innerText == "Show" ? "none" : "block"
+        document.getElementById(`user-${id}`).checked ? "block" : "none"
       );
     }
   }
 };
 
-new TomSelect("#modalUserId", {
-  maxItems: 3,
-  placeholder: "Select users",
-  search: true,
-  plugins: ["remove_button"],
-});
+// Update the staff's color
+window.updateColor = (id, color) => {
+  fetch("/admin/shifts/update_color", {
+    method: "POST",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      id: id,
+      color: color,
+      format: "json",
+    }),
+  })
+    .then((response) => {
+      if (response.ok) {
+        const toast = new bootstrap.Toast(
+          document.getElementById("toast-color-update-success")
```

Comment:
Use TurboLinks to reload the page `Turbolinks.visit(window.location, { action: "replace" });`